### PR TITLE
Serratus Analysis Index API calls and Link Buttons

### PR DESCRIPTION
**Pull request by ababaian**
Line 5-17 in `/src/components/Explorer/Base/Result/MatchingRuns/SerratusApiCalls.jsx`

```
export const getAnalysisIndexUrl = (run_id) => {
    const params = {
        run_id: run_id,
    }
    params[run_id] = run_id
    return `${baseUrl}/index/run=${run_id}`
}
export const getAnalysisMicro = async (run_id) => {
    const AnalysisUrl = getAnalysisIndexUrl(run_id)
    const response = await axios.get({ AnalysisUrl })
    return response
}
```

To the LinkButton added this in src/common/LinkButton.jsx I added option to hide
```
    if (show) {
        return (
            <button className='bg-gray-200 hover:bg-gray-400 mx-2 py-1 px-4 rounded inline-flex items-center'>
                <a className='text-blue-500' {...aAttrs} href={link} download={download}>
                    {text}
                    {icon}
                </a>
            </button>
        )
    } else {
        return null
    }
```

I can't get a proper API return
```
    const microAvailable = getAnalysisMicro(run_id)
```
This does not show the data return from the API
```
            <div>{microAvailable.data}</div>
```

### CODE REVIEW

File: src/components/Explorer/Rdrp/LinkButtons.jsx [hunk 1]
```diff
@@ -64,6 +66,12 @@ function RunLinkButtons({ run_id }) {
                 icon={externalLinkIcon}
                 newTab={true}
             />
+            <LinkButton
+                link={`${baseUrl}/explorer?run=${run_id}`}
+                text='NT'
+                icon={externalLinkIcon}
```
Comment (victorlin) [inline src/components/Explorer/Rdrp/LinkButtons.jsx:72]: remove `externalLinkIcon` if this isn't an external link

File: src/components/Explorer/Rdrp/LinkButtons.jsx [hunk 2]
```diff
@@ -49,6 +66,12 @@ function RunLinkButtons({ run_id }) {
                 icon={externalLinkIcon}
                 newTab={true}
             />
+            <LinkButton
+                link={`${baseUrl}/explorer?run=${run_id}`}
```
Comment (victorlin) [inline src/components/Explorer/Rdrp/LinkButtons.jsx:70]: if linking within the website, remove `baseUrl`:

```javascript
link={`/explorer?run=${run_id}`}
```

File: src/components/Explorer/Rdrp/LinkButtons.jsx [hunk 3]
```diff
@@ -35,6 +38,20 @@ function SequenceLinkButtons({ sequence_accession }) {
...
+    React.useEffect(() => {
+        async function setIndexButtons() {
+            const responseData = await getAnalysisIndex(run_id)
+            setMicroAvailable(responseData.analysis_index.micro)
+            setNtAvailable(responseData.analysis_index.nsra)
+        }
+        setIndexButtons()
+    }, [])
+
+    console.log('Hello world!')
```
Comment (victorlin) [inline src/components/Explorer/Rdrp/LinkButtons.jsx:54]: remove `console.log()`

File: src/components/Explorer/Rdrp/LinkButtons.jsx [hunk 4]
```diff
@@ -13,6 +14,8 @@ export const LinkButtons = ({ searchLevel, searchLevelValue }) => {
     }
 }
 
+export const AnalysisIndex = 'TRUE'
```
Comment (victorlin) [inline src/components/Explorer/Rdrp/LinkButtons.jsx:17]: remove unused var

File: src/components/Explorer/Base/Result/MatchingRuns/SerratusApiCalls.jsx [hunk 5]
```diff
@@ -2,6 +2,20 @@ import axios from 'axios'
...
+export const getAnalysisIndexUrl = (run_id) => {
+    const params = {
+        run_id: run_id,
+    }
+    params[run_id] = run_id
+    return `${baseUrl}/index/run=${run_id}`
+}
+
+export const getAnalysisIndex = async (run_id) => {
+    const AnalysisUrl = getAnalysisIndexUrl(run_id)
```
Comment (victorlin) [inline src/components/Explorer/Base/Result/MatchingRuns/SerratusApiCalls.jsx:14]: `getAnalysisIndexUrl()` is only used here. suggest merging the functions, dropping unused `params`, and using camelCase:

```javascript
export const getAnalysisIndex = async (run_id) => {
    const analysisUrl = `${baseUrl}/index/run=${run_id}`
    const response = await axios.get(analysisUrl)
    return response.data
}
```

File: src/common/LinkButton.jsx [hunk 6]
```diff
@@ -1,19 +1,30 @@
...
     }
-    return (
-        <button className='bg-gray-200 hover:bg-gray-400 mx-2 py-1 px-4 rounded inline-flex items-center'>
-            <a className='text-blue-500' {...aAttrs} href={link} download={download}>
-                {text}
-                {icon}
-            </a>
-        </button>
-    )
+    if (show) {
```
Comment (victorlin) [inline src/common/LinkButton.jsx:18]: suggest returning null first for less nesting:

```javascript
if (!show) return null

return <button ...
```

**Review by victorlin — CHANGES_REQUESTED** [review]

File: src/components/Explorer/Rdrp/LinkButtons.jsx [hunk 7]
```diff
@@ -35,6 +38,20 @@ function SequenceLinkButtons({ sequence_accession }) {
...
 
 function RunLinkButtons({ run_id }) {
+    const baseUrl = window.location.origin
+    const [microAvailable, setMicroAvailable] = React.useState(false)
+    const [ntAvailable, setNtAvailable] = React.useState(false)
+
+    React.useEffect(() => {
+        async function setIndexButtons() {
+            const responseData = await getAnalysisIndex(run_id)
+            setMicroAvailable(responseData.analysis_index.micro)
```
Comment (victorlin) [inline src/components/Explorer/Rdrp/LinkButtons.jsx:48]: I don't like the intermediate `analysis_index` here. Suggest we change the API to return without nesting:

```json
{
    "run_id": "ERR2756788",
    "srarun": true,
    "nsra": true,
    "psra": true,
    "rsra": true,
    "assembly": true,
    "micro": true,
    "geo": false
}
```

then we can use this:

```javascript
const analysisIndex = await getAnalysisIndex(run_id)
setMicroAvailable(analysisIndex.micro)
setNtAvailable(analysisIndex.nsra)
```

Comment (ababaian) [reply src/components/Explorer/Rdrp/LinkButtons.jsx:48]: Fixed it by changing the function return to `analysis_index`

Comment (ababaian) [reply src/components/Explorer/Rdrp/LinkButtons.jsx:48]:
```
export const getAnalysisIndex = async (run_id) => {
    const analysisUrl = `${baseUrl}/index/run=${run_id}`
    const response = await axios.get(analysisUrl)
    return response.data.analysis_index
}
```

**Review by victorlin — APPROVED** [review]

**Pull request merged**